packages/lodestar/src/executionEngine/http.ts

```diff
@@ -68,24 +68,62 @@ export class ExecutionEngineHttp implements IExecutionEngine {
   async executePayload(executionPayload: merge.ExecutionPayload): Promise<ExecutePayloadResponse> {
     const method = "engine_executePayloadV1";
     const serializedExecutionPayload = serializeExecutionPayload(executionPayload);
-    const {status, latestValidHash} = await this.rpc.fetch<
-      EngineApiRpcReturnTypes[typeof method],
-      EngineApiRpcParamTypes[typeof method]
-    >({
-      method,
-      params: [serializedExecutionPayload],
-    });
+    const {status, latestValidHash, validationError} = await this.rpc
+      .fetch<EngineApiRpcReturnTypes[typeof method], EngineApiRpcParamTypes[typeof method]>({
+        method,
+        params: [serializedExecutionPayload],
+      })
+      /**
+       * If there are errors by EL like connection refused, internal error, they need to be
+       * treated seperate from being INVALID. For now, just pass the error upstream.
+       */
+      .catch((e: Error) => ({status: ExecutePayloadStatus.ELERROR, latestValidHash: null, validationError: e.message}));
```

Comment:
This catch is too generic, it should be able to differentiate between network errors and legit internal EL errors

Reply:
Do you think they will need to be processed differently in the upstream `verifyBlockStateTransition` function? or we want the differentiation for being detailed/thorough?